preview-maker.js

```diff
@@ -1,184 +1,391 @@
-    ...get(node, className, `image`),
-    ...get(node, className, `tspan`),
-    ...get(node, className, `stop`),
+  ...get(node, className, `rect`),
+  ...get(node, className, `circle`),
+  ...get(node, className, `path`),
+  ...get(node, className, `g`),
+  ...get(node, className, `polygon`),
+  ...get(node, className, `image`),
+  ...get(node, className, `tspan`),
+  ...get(node, className, `stop`),
 ];
-
-const fill = (node, color) => {
-    const { red, green, blue, alpha } = color;
-    const rgba = `rgba(${red}, ${green}, ${blue}, ${alpha / 255})`;
-
-    if (node.tagName === `stop`) {
-        node.setAttribute(`stop-color`, rgba);
-    } else {
-        node.setAttribute(`fill`, rgba);
+function rgbToHsl(rgbArr) {
+  const hsl = snejugalColor(rgbArr);
+  const hslPreRound = {
+    hue: hsl.hue,
+    saturation: hsl.saturation * 100,
+    lightness: hsl.lightness * 100,
+    alpha: hsl.alpha,
+  };
+  const result = {
+    hue: Math.round(hslPreRound.hue),
+    saturation: Math.round(hslPreRound.saturation),
+    lightness: Math.round(hslPreRound.lightness),
+    alpha: hsl.alpha,
+  };
+  return result;
+}
+function findMostFrequentColors(colors) {
+  var colorsQuantity = {};
+  var max = 0;
+  var accentHue;
+  for (const colornow of colors) {
+    if (colornow != 0) {
+      if (colorsQuantity[colornow] == null) {
+        colorsQuantity[colornow] = 0;
+      }
+      colorsQuantity[colornow] += 1;
+      if (colorsQuantity[colornow] > max) {
+        accentHue = colornow;
+        max = colorsQuantity[colornow];
+      }
     }
-
-    if (node.childNodes) {
-        for (let child of Array.from(node.childNodes)) {
-            if (child.setAttribute) {
-                fill(child, color);
-            }
+  }
+  var colorsHue = [];
+  for (colornow in colorsQuantity) {
+    if (colorsQuantity[colornow] == max) {
+      colorsHue.push(colornow);
+    }
+  }
+  if (colorsHue.length > 1) {
+    var minDifference = 360;
+    for (var colorHue of colorsHue) {
+      for (var hue of colorsHue) {
+        if (hue != colorHue) {
+          hue = Number(hue);
+          colorHue = Number(colorHue);
+          if (Math.abs(hue - colorHue) < minDifference) {
+            minDifference = Math.abs(hue - colorHue);
+            accentHue = (hue + colorHue) / 2;
+          }
         }
+      }
+    }
+  }
+  return accentHue;
+}
+const fill = (node, color) => {
+  if (node.tagName === `stop`) {
+    node.setAttribute(`stop-color`, color);
+  } else {
+    node.setAttribute(`fill`, color);
+  }
+
+  if (node.childNodes) {
+    for (let child of Array.from(node.childNodes)) {
+      if (child.setAttribute) {
+        fill(child, color);
+      }
     }
+  }
 };
-
+const preFill = (node, color) => {
+  if (color.hue == undefined) {
+    fill(
+      node,
+      `rgba(${color.red}, ${color.green}, ${color.blue}, ${color.alpha / 255})`
+    );
+  } else {
+    fill(node, `hsl(${color.hue}, ${color.saturation}%, ${color.lightness}%)`);
+  }
+};
```

Comment:
```suggestion
const fill = (rootNode, color) => {
  const cssColor = `red` in color
    ? `rgba(${color.red}, ${color.green}, ${color.blue}, ${color.alpha / 255})`
    : `hsla(${color.hue}, ${color.saturation * 100}%, ${color.lightness * 100}%, ${color.alpha / 255})`);
    
    const innerFill = (node) => {
      // the `fill` function`'s body here
    };
    
    innerFill(rootNode);
};
```

Assuming the above `rgbToHsl` is removed in favor of the one from `@snejugal/color`.